src/@utils/order.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  approve,
+  Datatoken,
+  FreOrderParams,
+  OrderParams,
+  ProviderInstance
+} from '@oceanprotocol/lib'
+import { AssetExtended } from 'src/@types/AssetExtended'
+import Web3 from 'web3'
+import { getOceanConfig } from './ocean'
+import { TransactionReceipt } from 'web3-eth'
+import { getSiteMetadata } from './siteConfig'
+
+/**
+ * For pool you need to buy the datatoken beforehand, this always assumes you want to order the first service
+ * @param web3
+ * @param asset
+ * @param accountId
+ * @returns {TokenReceipt} receipt of the order
+ */
+export async function order(
+  web3: Web3,
+  asset: AssetExtended,
+  accountId: string
+): Promise<TransactionReceipt> {
+  const datatoken = new Datatoken(web3)
+  console.log('asset order', asset)
+  const config = getOceanConfig(asset.chainId)
+  const { appConfig } = getSiteMetadata()
+
+  const initializeData = await ProviderInstance.initialize(
+    asset.id,
+    asset.services[0].id,
+    0,
+    accountId,
+    asset.services[0].serviceEndpoint
+  )
+
+  const orderParams = {
+    consumer: accountId,
+    serviceIndex: 0,
+    _providerFees: initializeData.providerFee
+  } as OrderParams
+
+  switch (asset.accessDetails?.type) {
+    case 'fixed': {
+      // this approve implies that basetToken is the same as swap fee token
+      const totalCost =
+        asset.accessDetails.price + Number.parseFloat(appConfig.marketFee)
```

Comment:
is this calculation safe? Still don't know if all our `Decimal` conversion and checks are all still needed in the pool tab, but if they are we should calculate with `Decimal` whenever we do calculations

Reply:
it's safe market fee will be something like 0.2 so no problems there

Reply:
fixed just to be on the safe side